mage_ai/data_preparation/models/block.py

```diff
@@ -540,11 +540,20 @@ def get_outputs(self, sample_count=None):
                 sample_count=sample_count,
             )
             if type(data) is pd.DataFrame:
+                analysis = variable_manager.get_variable(
+                    self.pipeline.uuid,
+                    self.uuid,
+                    v,
+                    variable_type=VariableType.DATAFRAME_ANALYSIS,
+                )
+                stats = analysis.get('statistics', {})
+                column_types = analysis.get('metadata', {}).get('column_types', {})
                 data = dict(
                     sample_data=dict(
                         columns=data.columns.tolist(),
                         rows=data.to_numpy().tolist(),
                     ),
+                    shape=[stats.get('original_row_count'), len(column_types)],
```

Comment:
let's fallback to `stats.get('count')` since the new stats wasn't available in old analysis.